### ISSUE TITLE
Exception when trying to print image

### ISSUE THREAD

**Issue by LostLogic** [d2phap/ImageGlass#1156]
### System information:
 - Windows OS version: Windows Server 2016
 - ImageGlass version: 8.2.6.6

### To Reproduce
1. Open an image
2. Select print or Ctrl+Print
3. Crash

### Actual behavior:
When trying to print an image, ImageGlass throws an exception

> See the end of this message for details on invoking just-in-time (JIT) debugging instead of this dialog box.
> ************** Exception Text **************
> System.ComponentModel.Win32Exception (0x80004005): No application is associated with the specified file for this operation
>    at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
>    at ImageGlass.frmMain.<mnuMainPrint_Click>g__PrintImage|156_0(String filename)
>    at ImageGlass.frmMain.mnuMainPrint_Click(Object sender, EventArgs e)
>    at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
>    at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
>    at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
>    at System.Windows.Forms.Control.WndProc(Message& m)
>    at System.Windows.Forms.ToolStrip.WndProc(Message& m)
>    at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
> ************** Loaded Assemblies **************
> mscorlib
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3850.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
> ----------------------------------------
> ImageGlass
>     Assembly Version: 8.2.6.6
>     Win32 Version: 8.2.6.6
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.exe
> ----------------------------------------
> System
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3650.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
> ----------------------------------------
> ImageGlass.Services
>     Assembly Version: 8.2.7807.28883
>     Win32 Version: 8.2.7807.28883
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.Services.DLL
> ----------------------------------------
> ImageGlass.Base
>     Assembly Version: 1.4.0.0
>     Win32 Version: 1.4.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.Base.DLL
> ----------------------------------------
> ImageGlass.Settings
>     Assembly Version: 8.2.0.0
>     Win32 Version: 8.2.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.Settings.DLL
> ----------------------------------------
> System.Windows.Forms
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3835.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
> ----------------------------------------
> System.Drawing
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.2558.0 built by: NET471REL1
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
> ----------------------------------------
> ImageGlass.Library
>     Assembly Version: 8.2.7807.28883
>     Win32 Version: 8.2.7807.28883
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.Library.DLL
> ----------------------------------------
> System.Core
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3835.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
> ----------------------------------------
> ImageGlass.UI
>     Assembly Version: 8.0.7807.28884
>     Win32 Version: 8.0.7807.28884
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.UI.DLL
> ----------------------------------------
> ImageGlass.Heart
>     Assembly Version: 8.2.0.0
>     Win32 Version: 8.2.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.Heart.DLL
> ----------------------------------------
> System.Xml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.2612.0 built by: NET471REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
> ----------------------------------------
> Magick.NET-Q16-OpenMP-x64
>     Assembly Version: 7.24.0.0
>     Win32 Version: 7.24.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/Magick.NET-Q16-OpenMP-x64.DLL
> ----------------------------------------
> Magick.NET.Core
>     Assembly Version: 7.0.0.0
>     Win32 Version: 7.0.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/Magick.NET.Core.DLL
> ----------------------------------------
> Magick.NET.SystemDrawing
>     Assembly Version: 2.0.16.0
>     Win32 Version: 2.0.16.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/Magick.NET.SystemDrawing.DLL
> ----------------------------------------
> FileWatcherEx
>     Assembly Version: 1.3.0.0
>     Win32 Version: 1.3.0.0
>     CodeBase: file:///C:/Program%20Files/ImageGlass/FileWatcherEx.DLL
> ----------------------------------------
> ImageGlass.ImageBox
>     Assembly Version: 8.0.7807.28883
>     Win32 Version: 8.0.7807.28883
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.ImageBox.DLL
> ----------------------------------------
> ImageGlass.ImageListView
>     Assembly Version: 11.0.0.7
>     Win32 Version: 11.0.0.7
>     CodeBase: file:///C:/Program%20Files/ImageGlass/ImageGlass.ImageListView.DLL
> ----------------------------------------
> System.Configuration
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3630.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
> ----------------------------------------
> ************** JIT Debugging **************
> To enable just-in-time (JIT) debugging, the .config file for this application or computer (machine.config) must have the
> jitDebugging value set in the system.windows.forms section.
> The application must also be compiled with debugging enabled.
> For example:
> <configuration>
>     <system.windows.forms jitDebugging="true" />
> </configuration>
> When JIT debugging is enabled, any unhandled exception will be sent to the JIT debugger registered on the computer
> rather than be handled by this dialog box.

I suspect this is due to it calling for the Print Image dialogue instead of the regular Print. Due to the setup in Citrix, there is no handlers available for the specific call, resulting in the exception and inability to print images from ImageGlass.

### Expected behavior:
Opening of print dialogue, allowing for printing of image

### Screenshots:
![image](https://user-images.githubusercontent.com/7568950/139394486-0057d5d1-3504-4086-9ce8-71d8d0ddfa17.png)

### Additional context:
Additional possibilities for the crash might be it running in a Citrix context.
Please note: I've tested this on a Windows Server 2019 Citrix server, and it works fine there for some reason.

**Comment by d2phap** [comment]
ImageGlass use "Print Pictures" dialog of Windows for photo printing, as I know, Windows Server does not have this feature.
I will fix the error and add a message to notify user about this

**Comment by LostLogic** [comment]
Ah, of course! It relies on the old Windows Photo Viewer to print!

After some fiddling around with the registry, this fixed the issue. Please note that you must have the Windows Photo Viewer directory and files in C:\Program Files (Not x86)
Leaving this here in case someone else run into this issue.
Copy/paste the text below into a reg file to add it to your registry:

################

```reg
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\Applications\photoviewer.dll]

[HKEY_CLASSES_ROOT\Applications\photoviewer.dll\shell]

[HKEY_CLASSES_ROOT\Applications\photoviewer.dll\shell\print]
"NeverDefault"=""

[HKEY_CLASSES_ROOT\Applications\photoviewer.dll\shell\print\command]
@=hex(2):25,00,53,00,79,00,73,00,74,00,65,00,6d,00,52,00,6f,00,6f,00,74,00,25,\
  00,5c,00,53,00,79,00,73,00,74,00,65,00,6d,00,33,00,32,00,5c,00,72,00,75,00,\
  6e,00,64,00,6c,00,6c,00,33,00,32,00,2e,00,65,00,78,00,65,00,20,00,22,00,25,\
  00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,46,00,69,00,6c,00,65,00,73,00,\
  25,00,5c,00,57,00,69,00,6e,00,64,00,6f,00,77,00,73,00,20,00,50,00,68,00,6f,\
  00,74,00,6f,00,20,00,56,00,69,00,65,00,77,00,65,00,72,00,5c,00,50,00,68,00,\
  6f,00,74,00,6f,00,56,00,69,00,65,00,77,00,65,00,72,00,2e,00,64,00,6c,00,6c,\
  00,22,00,2c,00,20,00,49,00,6d,00,61,00,67,00,65,00,56,00,69,00,65,00,77,00,\
  5f,00,46,00,75,00,6c,00,6c,00,73,00,63,00,72,00,65,00,65,00,6e,00,20,00,25,\
  00,31,00,00,00

[HKEY_CLASSES_ROOT\Applications\photoviewer.dll\shell\print\DropTarget]
"Clsid"="{60fd46de-f830-4894-a628-6fa81bc0190d}"

[HKEY_CLASSES_ROOT\SystemFileAssociations\image\shell\print]

[HKEY_CLASSES_ROOT\SystemFileAssociations\image\shell\print\command]
@=hex(2):25,00,53,00,79,00,73,00,74,00,65,00,6d,00,52,00,6f,00,6f,00,74,00,25,\
  00,5c,00,53,00,79,00,73,00,74,00,65,00,6d,00,33,00,32,00,5c,00,72,00,75,00,\
  6e,00,64,00,6c,00,6c,00,33,00,32,00,2e,00,65,00,78,00,65,00,20,00,22,00,25,\
  00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,46,00,69,00,6c,00,65,00,73,00,\
  25,00,5c,00,57,00,69,00,6e,00,64,00,6f,00,77,00,73,00,20,00,50,00,68,00,6f,\
  00,74,00,6f,00,20,00,56,00,69,00,65,00,77,00,65,00,72,00,5c,00,50,00,68,00,\
  6f,00,74,00,6f,00,56,00,69,00,65,00,77,00,65,00,72,00,2e,00,64,00,6c,00,6c,\
  00,22,00,2c,00,20,00,49,00,6d,00,61,00,67,00,65,00,56,00,69,00,65,00,77,00,\
  5f,00,46,00,75,00,6c,00,6c,00,73,00,63,00,72,00,65,00,65,00,6e,00,20,00,25,\
  00,31,00,00,00

[HKEY_CLASSES_ROOT\SystemFileAssociations\image\shell\print\DropTarget]
"Clsid"="{60fd46de-f830-4894-a628-6fa81bc0190d}"
```

**Comment by d2phap** [comment]
Awesome!

**Comment by d2phap** [comment]
I just implemented the new way to open Print Pictures in the latest build of ImageGlass Moon: https://imageglass.org/moon
If possible, could you help to test if the error still occurs?
Thanks!

**Comment by LostLogic** [comment]
Removed the reg-fix and the changes you've made in Moon seems to be working great! Sorry for not checking this earlier.

To be 100% sure, I launched ImageGlass first, and got the error mentioned in the initial post. Removed ImageGlass, then installed ImageGlass Moon. Moon had no errors, and popped up with the print dialogue shown below.

![image](https://user-images.githubusercontent.com/7568950/140699017-9bee67da-19db-4d46-a856-d2f96bd819c5.png)
